# 2) Variables (ตัวแปร) สำหรับงานวิเคราะห์ข้อมูลเบื้องต้น

บทนี้จะพาคุณเรียน “ตัวแปร (variables)” ซึ่งเป็นพื้นฐานสำคัญมากสำหรับงาน data  
เพราะการวิเคราะห์ข้อมูลมักต้อง **เก็บค่า** ไว้ใช้ซ้ำ เช่น ผลรวม ค่าเฉลี่ย จำนวนแถว ชื่อคอลัมน์ ฯลฯ

---

## สิ่งที่คุณจะได้จากบทนี้
- เข้าใจว่า **ตัวแปร = ชื่อที่ผูกกับค่า** เพื่อเรียกใช้ซ้ำได้
- ตั้งชื่อตัวแปรให้สื่อความหมายกับงานข้อมูลได้
- อัปเดตค่า (update) และตรวจสอบค่าได้
- เชื่อมกับงานวิเคราะห์ข้อมูลแบบง่าย ๆ (sum / average / counts)

---

## ภาพรวมเนื้อหา (ลำดับการเรียน)
1. Variables เทียบกับ Excel (ชื่อช่วง/ค่าในเซลล์)
2. การสร้างตัวแปรและเรียกใช้
3. การอัปเดตค่า (update)
4. การตรวจสอบค่า (print / output)
5. แบบฝึกหัดสั้น ๆ ในสไตล์งาน data

## A. Variables เทียบกับ Excel

ใน Excel คุณอาจ:
- เก็บข้อมูลไว้ใน **เซลล์** เช่น B2 = 305
- ใช้สูตรอ้างอิงเซลล์ เช่น `=B2/B3`
- หรือบางครั้งตั้งชื่อช่วงข้อมูล (Named Range) เพื่อเรียกใช้สะดวก

ใน Python เราทำแบบเดียวกันด้วย “ตัวแปร”  
✅ ตัวแปรคือ **ชื่อ** ที่เราให้กับ **ค่า** เพื่อเรียกใช้ซ้ำ

ตัวอย่าง:
- `total_score` เก็บคะแนนรวม
- `n_students` เก็บจำนวนนักเรียน

## B. สร้างตัวแปร (Assign value)

รูปแบบพื้นฐานคือ:
- `ชื่อตัวแปร = ค่า`

✅ ลองรัน cell ด้านล่าง

In [1]:
total_score = 305
n_students = 4

total_score

305

สังเกตว่า Jupyter จะแสดงค่า `total_score` ใต้ cell (output)  
เหมือนเรา “ดูค่าที่อยู่ในเซลล์” ใน Excel

## C. ใช้ตัวแปรเพื่อคำนวณ (เหมือนสูตรใน Excel)

ใน Excel: `=คะแนนรวม/จำนวนนักเรียน`  
ใน Python เราเขียนได้ตรง ๆ โดยใช้ตัวแปร

✅ ลองรัน cell นี้

In [2]:
average_score = total_score / n_students
average_score

76.25

## D. แสดงผลให้อ่านง่ายด้วย print()

บางครั้งเราอยากให้ผลลัพธ์อ่านง่ายขึ้น เช่น “คะแนนเฉลี่ย = 76.25”

✅ ลองรัน

In [3]:
print("คะแนนทั้งหมด =", total_score)
print("จำนวนนักเรียนทั้งหมด =", n_students)
print("คะแนนเฉลี่ย =", average_score)

คะแนนทั้งหมด = 305
จำนวนนักเรียนทั้งหมด = 4
คะแนนเฉลี่ย = 76.25


## E. การอัปเดตค่า (Update variable)

ในงาน data เรามัก:
- เพิ่มจำนวนข้อมูลเมื่ออ่านเพิ่ม
- เพิ่มผลรวมเมื่อรวมทีละส่วน
- แก้ค่าหลังตรวจสอบความถูกต้อง

ตัวอย่าง: ถ้าคะแนนรวมเพิ่มขึ้นอีก 10 คะแนน  
เราสามารถอัปเดตตัวแปรได้

✅ ลองรัน

In [4]:
total_score = total_score + 10
total_score

315

แล้วคำนวณค่าเฉลี่ยใหม่อีกครั้ง (เพราะค่ามีการเปลี่ยน)

In [5]:
average_score = total_score / n_students
average_score

78.75

## F. ตั้งชื่อตัวแปรให้เหมาะกับงานข้อมูล

การตั้งชื่อที่ดีทำให้อ่านโค้ดเหมือนอ่านรายงาน

✅ แนวทางง่าย ๆ
- ใช้ตัวพิมพ์เล็กและขีดล่าง เช่น `total_sales`, `avg_income`
- ชื่อสื่อความหมายกับข้อมูล
- หลีกเลี่ยงชื่อสั้นเกินไป เช่น `a`, `b` (ยกเว้นทดลองเร็ว ๆ)

ลองดูตัวอย่างที่ “อ่านแล้วเข้าใจทันที”:

In [6]:
total_sales = 125000
n_customers = 50
avg_sales_per_customer = total_sales / n_customers

print("Avg sales per customer =", avg_sales_per_customer)

Avg sales per customer = 2500.0


## G. แบบฝึกหัด 1: ค่าใช้จ่ายเฉลี่ยต่อวัน

สถานการณ์ (แบบงานจริง):
- คุณมีค่าใช้จ่ายรวมของสัปดาห์ (7 วัน)
- ต้องการหา “ค่าใช้จ่ายเฉลี่ยต่อวัน”

✅ ทำตามขั้นตอน
1. สร้างตัวแปร `total_expense`
2. สร้างตัวแปร `n_days`
3. คำนวณ `avg_expense_per_day`
4. แสดงผลด้วย `print()`

✍️ แก้ไขตัวเลขตามต้องการ แล้วรัน

In [7]:
total_expense = 3500
n_days = 7

avg_expense_per_day = total_expense / n_days
print("Average expense per day =", avg_expense_per_day)

Average expense per day = 500.0


## H. แบบฝึกหัด 2: ตรวจสอบความสมเหตุสมผลของค่า

ในงาน data เรามักตรวจสอบ “ค่าที่ได้สมเหตุสมผลไหม”  
เช่น จำนวนคนต้องเป็นจำนวนเต็ม และไม่ควรเป็น 0

✅ ลองแก้ค่าด้านล่าง แล้วรันดูว่าเกิดอะไรขึ้น
- ถ้า `n_students = 0` จะเกิดอะไร?

In [8]:
total_score = 305
n_students = 4

average_score = total_score / n_students
print("Average score =", average_score)

Average score = 76.25


## I. สรุป (Checklist)

หลังจบบทนี้ คุณควรทำได้:
- ✅ สร้างตัวแปรเพื่อเก็บค่าที่เกี่ยวกับข้อมูล
- ✅ ใช้ตัวแปรคำนวณเหมือนเขียนสูตรใน Excel
- ✅ อัปเดตค่าเมื่อข้อมูลเปลี่ยน
- ✅ แสดงผลให้อ่านง่ายด้วย `print()`
- ✅ ตั้งชื่อตัวแปรให้สื่อความหมาย

➡️ พร้อมเข้าสู่บทถัดไป: **Basic Data Types (ชนิดข้อมูลพื้นฐาน)** เพื่อเข้าใจตัวเลข/ข้อความ/จริง-เท็จ ในงาน data